### ISSUE TITLE
Raspberry PI 3 make error

### ISSUE THREAD

**Issue by zatmar** [richardjoo/HTMLCOIN#4]
I followed your guide to install the wallet in Ubuntu VPS (1gb ram) and I get this error when I compiled it

```make
Making all in src
make[1]: Entering directory `/root/Desktop/HTMLCOIN/src'
make[2]: Entering directory `/root/Desktop/HTMLCOIN/src'
  CXX      crypto/libbitcoinconsensus_la-aes.lo
  CXX      crypto/libbitcoinconsensus_la-chacha20.lo
  CXX      crypto/libbitcoinconsensus_la-hmac_sha256.lo
  CXX      crypto/libbitcoinconsensus_la-hmac_sha512.lo
  CXX      crypto/libbitcoinconsensus_la-ripemd160.lo
  CXX      crypto/libbitcoinconsensus_la-sha1.lo
  CXX      crypto/libbitcoinconsensus_la-sha256.lo
  CXX      crypto/libbitcoinconsensus_la-sha512.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-b64.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto-mcf.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto_scrypt-check.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto_scrypt-hash.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto_scrypt-hexconvert.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto_scrypt-nosse.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-crypto-scrypt-saltgen.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-sha256.lo
  CC       cpp-ethereum/utils/libscrypt/libbitcoinconsensus_la-slowequals.lo
  CXX      libbitcoinconsensus_la-arith_uint256.lo
  CXX      consensus/libbitcoinconsensus_la-merkle.lo
  CXX      libbitcoinconsensus_la-hash.lo
  CXX      primitives/libbitcoinconsensus_la-block.lo
  CXX      primitives/libbitcoinconsensus_la-transaction.lo
  CXX      libbitcoinconsensus_la-pubkey.lo
  CXX      script/libbitcoinconsensus_la-bitcoinconsensus.lo
  CXX      script/libbitcoinconsensus_la-interpreter.lo
  CXX      script/libbitcoinconsensus_la-script.lo
  CXX      script/libbitcoinconsensus_la-script_error.lo
  CXX      libbitcoinconsensus_la-uint256.lo
  CXX      libbitcoinconsensus_la-utilstrencodings.lo
make[3]: Entering directory `/root/Desktop/HTMLCOIN/src/secp256k1'
gcc -I. -g -O2 -Wall -Wextra -Wno-unused-function -c src/gen_context.c -o gen_context.o
gcc gen_context.o -o gen_context
./gen_context
  CC       src/libsecp256k1_la-secp256k1.lo
  CCLD     libsecp256k1.la
make[3]: Leaving directory `/root/Desktop/HTMLCOIN/src/secp256k1'
  CXXLD    libbitcoinconsensus.la
copying selected object files to avoid basename conflicts...
  CXX      htmlcoind-bitcoind.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-blockencodings.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-checkpointsync.o
checkpointsync.cpp: In member function ‘bool CSyncCheckpoint::ProcessSyncCheckpoint(CNode*)’:
checkpointsync.cpp:349:15: warning: unused variable ‘pindexCheckpoint’ [-Wunused-variable]
  CBlockIndex* pindexCheckpoint = mapBlockIndex[hashCheckpoint];
               ^
  CXX      libbitcoin_server_a-httprpc.o
  CXX      libbitcoin_server_a-httpserver.o
  CXX      libbitcoin_server_a-init.o
  CXX      libbitcoin_server_a-dbwrapper.o
  CXX      libbitcoin_server_a-merkleblock.o
  CXX      libbitcoin_server_a-miner.o
  CXX      libbitcoin_server_a-net.o
  CXX      libbitcoin_server_a-net_processing.o
  CXX      libbitcoin_server_a-noui.o
  CXX      policy/libbitcoin_server_a-fees.o
  CXX      policy/libbitcoin_server_a-policy.o
  CXX      libbitcoin_server_a-pow.o
  CXX      libbitcoin_server_a-pos.o
  CXX      libbitcoin_server_a-rest.o
  CXX      rpc/libbitcoin_server_a-blockchain.o
  CXX      rpc/libbitcoin_server_a-mining.o
  CXX      rpc/libbitcoin_server_a-misc.o
  CXX      rpc/libbitcoin_server_a-net.o
  CXX      rpc/libbitcoin_server_a-rawtransaction.o
  CXX      rpc/libbitcoin_server_a-server.o
  CXX      script/libbitcoin_server_a-sigcache.o
  CXX      script/libbitcoin_server_a-ismine.o
  CXX      libbitcoin_server_a-timedata.o
  CXX      libbitcoin_server_a-torcontrol.o
  CXX      libbitcoin_server_a-txdb.o
  CXX      libbitcoin_server_a-txmempool.o
  CXX      libbitcoin_server_a-ui_interface.o
  CXX      libbitcoin_server_a-validation.o
validation.cpp: In function ‘bool ContextualCheckBlockHeader(const CBlockHeader&, CValidationState&, const Consensus::Params&, con                              st CBlockIndex*, int64_t)’:
validation.cpp:4093:15: warning: unused variable ‘nHeight’ [-Wunused-variable]
     const int nHeight = pindexPrev == NULL ? 0 : pindexPrev->nHeight + 1;
               ^
  CXX      libbitcoin_server_a-validationinterface.o
  CXX      libbitcoin_server_a-versionbits.o
  CXX      qtum/libbitcoin_server_a-qtumstate.o
  CXX      qtum/libbitcoin_server_a-qtumtransaction.o
  CXX      qtum/libbitcoin_server_a-qtumDGP.o
  CXX      consensus/libbitcoin_server_a-consensus.o
  CXX      qtum/libbitcoin_server_a-storageresults.o
  CXX      support/libbitcoin_server_a-allocators.o
  AR       libbitcoin_server.a
  CXX      libbitcoin_common_a-amount.o
  CXX      libbitcoin_common_a-base58.o
  CXX      libbitcoin_common_a-chainparams.o
  CXX      libbitcoin_common_a-coins.o
  CXX      libbitcoin_common_a-compressor.o
  CXX      libbitcoin_common_a-core_read.o
  CXX      libbitcoin_common_a-core_write.o
  CXX      libbitcoin_common_a-key.o
  CXX      libbitcoin_common_a-keystore.o
  CXX      libbitcoin_common_a-netaddress.o
  CXX      libbitcoin_common_a-netbase.o
  CXX      libbitcoin_common_a-opensslkey.o
  CXX      libbitcoin_common_a-protocol.o
  CXX      libbitcoin_common_a-scheduler.o
  CXX      script/libbitcoin_common_a-sign.o
  CXX      script/libbitcoin_common_a-standard.o
  CXX      libbitcoin_common_a-warnings.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Base64.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Common.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-CommonData.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-CommonIO.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-CommonJS.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-FileSystem.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-FixedHash.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Guards.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Hash.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Log.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-MemoryDB.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-OverlayDB.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-RLP.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-SHA3.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-TransientDirectory.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-TrieCommon.o
  CXX      cpp-ethereum/libdevcore/libbitcoin_common_a-Worker.o
  CXX      cpp-ethereum/libevm/libbitcoin_common_a-ExtVMFace.o
  CXX      cpp-ethereum/libevm/libbitcoin_common_a-VM.o
  CXX      cpp-ethereum/libevm/libbitcoin_common_a-VMOpt.o
  CXX      cpp-ethereum/libevm/libbitcoin_common_a-VMCalls.o
In file included from /usr/include/boost/optional.hpp:15:0,
                 from /usr/include/boost/format/internals.hpp:20,
                 from /usr/include/boost/format.hpp:38,
                 from /usr/include/boost/math/policies/error_handling.hpp:31,
                 from /usr/include/boost/multiprecision/detail/default_ops.hpp:9,
                 from /usr/include/boost/multiprecision/detail/generic_interconvert.hpp:9,
                 from /usr/include/boost/multiprecision/number.hpp:22,
                 from /usr/include/boost/multiprecision/cpp_int.hpp:12,
                 from ./cpp-ethereum/libdevcore/Common.h:51,
                 from ./cpp-ethereum/libethcore/Common.h:28,
                 from ./cpp-ethereum/libethereum/ExtVM.h:26,
                 from cpp-ethereum/libevm/VMCalls.cpp:19:
/usr/include/boost/optional/optional.hpp: In instantiation of ‘void boost::optional_detail::optional_base<T>::construct(boost::opt                              ional_detail::optional_base<T>::argument_type) [with T = dev::eth::owning_bytes_ref; boost::optional_detail::optional_base<T>::arg                              ument_type = const dev::eth::owning_bytes_ref&]’:
/usr/include/boost/optional/optional.hpp:250:33:   required from ‘boost::optional_detail::optional_base<T>::optional_base(const bo                              ost::optional_detail::optional_base<T>&) [with T = dev::eth::owning_bytes_ref]’
/usr/include/boost/optional/optional.hpp:564:76:   required from ‘boost::optional<T>::optional(const boost::optional<T>&) [with T                               = dev::eth::owning_bytes_ref]’
cpp-ethereum/libevm/VMCalls.cpp:137:68:   required from here
/usr/include/boost/optional/optional.hpp:346:8: error: use of deleted function ‘dev::eth::owning_bytes_ref::owning_bytes_ref(const                               dev::eth::owning_bytes_ref&)’
        new (m_storage.address()) internal_type(val) ;
        ^
In file included from ./cpp-ethereum/libethereum/ExtVM.h:27:0,
                 from cpp-ethereum/libevm/VMCalls.cpp:19:
./cpp-ethereum/libevm/ExtVMFace.h:72:2: error: declared here
  owning_bytes_ref(owning_bytes_ref const&) = delete;
  ^
make[2]: *** [cpp-ethereum/libevm/libbitcoin_common_a-VMCalls.o] Error 1
make[2]: Leaving directory `/root/Desktop/HTMLCOIN/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/Desktop/HTMLCOIN/src'
make: *** [all-recursive] Error 1

```

**Comment by richardjoo** [comment]
* did you use ubuntu mate? raspbian will not work
* did you make sure you created swap and restarted your rpi3?
* make sure to run `sudo apt-get update && sudo apt-get upgrade`
* rpi3 installation is tough for sure.  It took me several attempts to make this happen.

Please let me know if everything was properly followed.  I hope you and I can find the issue and update the instruction :-)

**Comment by zatmar** [comment]
Sorry, I was wrong to write the issue.
I follow your raspberry tutorial for install HTML wallet in a low vps (1gb RAM) with Ubuntu.
First I received a memory error when installing the normal wallet, but when I followed their Raspberry PI3 guide and created the exchange, I fixed this old error.

But now I have this error. I using Ubuntu 14.04. If we can fix that flaw, I'll be able to do a quick guide on how to mine HTML at a vps for 1 dollar a month.

**Comment by richardjoo** [comment]
I have not tested this with ubuntu 14.04 on rpi3.  Only one I was able to make this work was with 16.04.2 (Xenial).

14.04 on regular machine will work just fine, but architecture like rpi3 (aarch32 (ARMv7)) may not work.

Please let me know :-)

**Comment by zatmar** [comment]
Fixed when upgrading to ubuntu 16.04. I'm going to write the guide. A greeting!
